app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/adapters/recycleradapters/FeedbackAdapter.kt

```diff
@@ -54,31 +54,29 @@ class FeedbackAdapter(
             arrangedFeedbackList.clear()
             feedbackResponse.feedbackList.forEach { item ->
                 if (item.email == PrefManager.getStringSet(Constant.SAVED_EMAIL)?.iterator()?.next()) {
-                    var feedback = Feedback()
-                    feedback.email = item.email
-                    feedback.userName = item.userName
-                    feedback.avatar = item.avatar
-                    feedback.timestamp = item.timestamp
-                    feedback.feedback = item.feedback
-                    arrangedFeedbackList.add(feedback)
+                    addFeedback(item)
                 }
             }
 
             var reverseResponse = feedbackResponse.feedbackList.asReversed()
             reverseResponse.forEach { item ->
                 if (item.email != PrefManager.getStringSet(Constant.SAVED_EMAIL)?.iterator()?.next()) {
-                    var feedback = Feedback()
-                    feedback.email = item.email
-                    feedback.userName = item.userName
-                    feedback.avatar = item.avatar
-                    feedback.timestamp = item.timestamp
-                    feedback.feedback = item.feedback
-                    arrangedFeedbackList.add(feedback)
+                    addFeedback(item)
                 }
             }
         Timber.d("Arranged the feedback")
     }
 
+    private fun addFeedback(item: Feedback) {
+        var feedback = Feedback()
+        feedback.email = item.email
+        feedback.userName = item.userName
+        feedback.avatar = item.avatar
+        feedback.timestamp = item.timestamp
+        feedback.feedback = item.feedback
```

Comment:
Why are you copying it?

Reply:
Currently what we have is that we get a response of the feedback in the order of newest at the top. But what we want is that, if the user logged in has given any feedback, then it should come at the top irrespective of the time he gave that feedback. After user feedback is shown, we need to show the rest of the feedbacks in the order with the newest at the top. So, to arrange the list in such an order, I am keeping a separate list of those items and showing these data, with actually affecting the server response directly.

Reply:
Why are you creating a new feedback object and copying the same data in it?

Reply:
The actual feedback response has many different kinds of data, we don't need those data to display or even to iterate any operation. So, I made a feedback object, which can copy the response that we need, instead of copying all the data.

Reply:
If you don't create a new object, will there be any problem?

Reply:
No, what I can do is that I can create the object only once and reuse it every time.

Reply:
Why do you event want to create an object?

Reply:
We don't need. I understood what u wanted to say. Updating the pr accordingly.